Packet++/src/HttpLayer.cpp

```diff
@@ -890,30 +847,32 @@ void HttpResponseFirstLine::setVersion(HttpVersion newVersion)
 	m_Version = newVersion;
 }
 
-HttpResponseLayer::HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
+HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
 {
 	// minimum data should be 12B long: "HTTP/x.y XXX"
 	if (!data || dataLen < 12)
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	const char* statusCodeData = data + 9;
+	std::string codeString = std::string(data + 9, 3);
 
-	auto codeAsEnum = StatusCodeStringToEnumMap.find(std::string(statusCodeData, 3));
-	if (codeAsEnum == StatusCodeStringToEnumMap.end())
+	if(!codeString.empty() && std::find_if(codeString.begin(), codeString.end(), [](unsigned char c){ return !std::isdigit(c); }) == codeString.end())
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
-	return codeAsEnum->second;
+
+	// cppcheck-suppress containerOutOfBounds
+	int statusCodeInt = codeString[0] * 100 + codeString[1] * 10 + codeString[2];
```

Comment:
This code doesn't work, because `codeString[x]` is of type `char` so multiplying it by a number will give the ascii value x the number. For example: if `codeString[0] == '1'` then `codeString[0] * 100 == 4900` (because the ascii value of `'1'` is `49`).

Here it's actually safe to use `std::stoi(codeString)` because we know the string only contains digits. It will also help us get rid of `// cppcheck-suppress containerOutOfBounds`

We can just do:
```cpp
return HttpResponseStatusCode(std:stoi(codeString));
```

BTW, not sure if you saw but the tests fail in CI:
https://github.com/seladb/PcapPlusPlus/actions/runs/5963344911?pr=1180

Reply:
ok, I see your point, you want to check the string first and call `std:stoi`

Reply:
fixed. now, it check if it is a number first and then call `stoi`